remote_attestation/rust/src/message.rs

```diff
@@ -26,16 +27,36 @@ use serde_big_array::BigArray;
 const CLIENT_HELLO_HEADER: u8 = 1;
 const SERVER_IDENTITY_HEADER: u8 = 2;
 const CLIENT_IDENTITY_HEADER: u8 = 3;
+const ENCRYPTED_DATA_HEADER: u8 = 4;
 
-/// Remote Attestation protocol version.
+/// Remote attestation protocol version.
 pub const PROTOCOL_VERSION: u8 = 1;
 
 /// Length (in bytes) of the random vector sent in messages for preventing replay attacks.
 pub const REPLAY_PROTECTION_ARRAY_LENGTH: usize = 32;
 
+/// Convenience struct that wraps attestation messages.
+pub enum MessageWrapper {
```

Comment:
Isn't this MessageWrapper exactly what I mentioned in the other comment as the high level frame? In fact, I think the parsing and serialization should be done via this type, not the individual message types.